data/transactions/logic/assembler.go

```diff
@@ -1199,17 +1204,22 @@ func (ops *OpStream) pragma(line string) error {
 		if ver < 1 || ver > AssemblerMaxVersion {
 			return ops.errorf("unsupported version: %d", ver)
 		}
+		if ops.pragraVersionFound {
+			return ops.errorf("pragma version can appear only once in a TEAL program")
+		}
 
 		// We initialize Version with assemblerNoVersion as a marker for
 		// non-specified version because version 0 is valid
 		// version for TEAL v1.
 		if ops.Version == assemblerNoVersion {
 			ops.Version = ver
-		} else if ops.Version != ver {
-			return ops.errorf("version mismatch: assembling v%d with v%d assembler", ops.Version, ver)
+		} else if ops.Version < ver {
```

Comment:
No, it was the exact opposite. the `tealdbg` started failed after adding `LogicSigVersion = 3`, since the default version was higher than the one provided in the pragma directive. I think that for our tooling, there is no reason we shouldn't be able to support the old versions as well. Adding a protocol specific argument to the command line is a reasonable way to set the teal *default* version.